File: 1614_max_nesting_depths/MaxNestingDepths.java
class Solution {
    public int maxDepth(String s) {
        int maxNesting = 0;
        int currentParentheses = 0;

        for (int i : s.toCharArray()) {
            if (i == '(') {
                currentParentheses++;
                maxNesting = Math.max(currentParentheses, maxNesting);
            }
            if (i == ')') {
                currentParentheses--;
            }
        }

        return maxNesting;
    }
}

File: 1614_max_nesting_depths/max_nesting_depths.py
# With stack

class Solution:
    def maxDepth(self, s: str) -> int:
        stack = []
        maxNesting = 0


        for i in s:
            if i == '(':
                stack.append(i)
            if i == ')':
                stack.pop()

            n = len(stack)

            if n > maxNesting:
                maxNesting = n


        return maxNesting